cmd/restic/cmd_backup.go

```diff
@@ -427,6 +429,18 @@ func runBackup(opts BackupOptions, gopts GlobalOptions, args []string) error {
 			return false
 		}
 
+		if opts.ExcludeCaches && fi.Name() != "CACHEDIR.TAG" {
+			d := filepath.Dir(item)
+			_, err := os.Stat(filepath.Join(d, "CACHEDIR.TAG"))
```

Comment:
Oh, right, I forgot about the header. I should have read the relevant section of `tar(1)` instead of only the excerpt in #317. 😁 
I fully agree that it's best to refactor the check into a function and will proceed accordingly.